stores/milvus-documentstore/milvus_documentstore/milvus.py

```diff
@@ -127,269 +132,307 @@ def __init__(
+                        Filters are defined as nested dictionaries. The keys of the dictionaries can be a logical
+                        operator (`"$and"`, `"$or"`, `"$not"`), a comparison operator (`"$eq"`, `"$in"`, `"$gt"`,
+                        `"$gte"`, `"$lt"`, `"$lte"`) or a metadata field name.
+                        Logical operator keys take a dictionary of metadata field names and/or logical operators as
+                        value. Metadata field names take a dictionary of comparison operators as value. Comparison
+                        operator keys take a single value or (in case of `"$in"`) a list of values as value.
+                        If no logical operator is provided, `"$and"` is used as default operation. If no comparison
+                        operator is provided, `"$eq"` (or `"$in"` if the comparison value is a list) is used as default
+                        operation.
+
+                            __Example__:
+
+                            ```python
+                            filters = {
+                                "$and": {
+                                    "type": {"$eq": "article"},
+                                    "date": {"$gte": "2015-01-01", "$lt": "2021-01-01"},
+                                    "rating": {"$gte": 3},
+                                    "$or": {
+                                        "genre": {"$in": ["economy", "politics"]},
+                                        "publisher": {"$eq": "nytimes"}
+                                    }
+                                }
+                            }
+                            ```
+
+        :param return_embedding: Whether to return the document embeddings.
+        :param batch_size: Number of documents that are passed to bulk function at a time.
+        :param headers: Custom HTTP headers to pass to document store client if supported (e.g. {'Authorization': 'Basic YWRtaW46cm9vdA=='} for basic authentication)
+        """
+        index = index or self.collection_name
 
-        index = index or self.index
-        index_param = index_param or self.index_param
-        duplicate_documents = duplicate_documents or self.duplicate_documents
-        assert (
-            duplicate_documents in self.duplicate_documents_options
-        ), f"duplicate_documents parameter must be {', '.join(self.duplicate_documents_options)}"
-        field_map = self._create_document_field_map()
+        logger.warn("Batch_size not supported, loading all results.")
 
-        if len(documents) == 0:
-            logger.warning("Calling DocumentStore.write_documents() with empty list")
-            return
+        filters = (
+            LogicalFilterClause.parse(filters).convert_to_milvus() if filters else None
+        )
+        print(filters)
+        res = self.client.query(
+            collection_name=index,
+            filter_expression=filters,
+            include_vectors=return_embedding,
```

Comment:
In Haystack's document stores, the `return_embedding` parameter is usually set globally when initializing the document store. The global behavior can be changed on a per case distinction in the different through the optional `return_embeddings` parameter.